src/19FAF0.c

```diff
@@ -220,11 +463,213 @@ ApiStatus GetPlayerHP(Evt* script, s32 isInitialCall) {
+    } else {
+        gBattleStatus.flags1 &= ~BS_FLAGS1_40;
+    }
+    if (flags & BS_FLAGS1_200) {
+        gBattleStatus.flags1 |= BS_FLAGS1_200;
+    } else {
+        gBattleStatus.flags1 &= ~BS_FLAGS1_200;
+    }
+    if (flags & BS_FLAGS1_80) {
+        gBattleStatus.flags1 |= BS_FLAGS1_80;
+    } else {
+        gBattleStatus.flags1 &= ~BS_FLAGS1_80;
+    }
+    if (flags & BS_FLAGS1_800) {
+        gBattleStatus.flags1 |= BS_FLAGS1_800;
+    } else {
+        gBattleStatus.flags1 &= ~BS_FLAGS1_800;
+    }
+
+    target = get_actor(script->owner1.actorID);
+    battleStatus->currentTargetID = target->targetActorID;
+    battleStatus->currentTargetPart = target->targetPartIndex;
+    battleStatus->statusChance = battleStatus->currentAttackStatus;
+    if (battleStatus->statusChance == 0xFF) {
+        battleStatus->statusChance = 0;
+    }
+    battleStatus->statusDuration = (battleStatus->currentAttackStatus & 0xF00) >> 8;
+
+    temp_v0_2 = calc_player_damage_enemy();
+    if (temp_v0_2 < 0) {
+        return ApiStatus_FINISH;
+    }
+    evt_set_variable(script, a0, temp_v0_2);
+
+    if (!does_script_exist_by_ref(script)) {
+        return ApiStatus_FINISH;
+    }
+
+    return ApiStatus_DONE2;
+}
+
+ApiStatus PlayerTestEnemy(Evt* script, s32 isInitialCall) {
+    BattleStatus* battleStatus = &gBattleStatus;
+    Bytecode* args = script->ptrReadPos;
+    s32 a0 = *args++;
+    s32 flags;
+    Actor* target;
+    s32 temp_v0_2;
+
+    battleStatus->currentAttackElement = *args++;
+    battleStatus->currentAttackEventSuppression = *args++;
+    battleStatus->currentAttackStatus = *args++;
+    battleStatus->currentAttackDamage = evt_get_variable(script, *args++);
+    battleStatus->powerBounceCounter = 0;
+    flags = *args++;
+
+    if ((flags & (BS_FLAGS1_10 | BS_FLAGS1_SP_EVT_ACTIVE)) == (BS_FLAGS1_10 | BS_FLAGS1_SP_EVT_ACTIVE)) {
+        gBattleStatus.flags1 |= BS_FLAGS1_10 | BS_FLAGS1_SP_EVT_ACTIVE;
+    } else if (flags & BS_FLAGS1_10) {
+        gBattleStatus.flags1 |= BS_FLAGS1_10;
+        gBattleStatus.flags1 &= ~BS_FLAGS1_SP_EVT_ACTIVE;
+    } else if (flags & BS_FLAGS1_SP_EVT_ACTIVE) {
+        gBattleStatus.flags1 &= ~BS_FLAGS1_10;
+        gBattleStatus.flags1 |= BS_FLAGS1_SP_EVT_ACTIVE;
+    } else {
+        gBattleStatus.flags1 &= ~BS_FLAGS1_10;
+        gBattleStatus.flags1 &= ~BS_FLAGS1_SP_EVT_ACTIVE;
+    }
+
+    if (flags & BS_FLAGS1_40) {
+        gBattleStatus.flags1 |= BS_FLAGS1_40;
+    } else {
+        gBattleStatus.flags1 &= ~BS_FLAGS1_40;
+    }
+    if (flags & BS_FLAGS1_200) {
+        gBattleStatus.flags1 |= BS_FLAGS1_200;
+    } else {
+        gBattleStatus.flags1 &= ~BS_FLAGS1_200;
+    }
+    if (flags & BS_FLAGS1_80) {
+        gBattleStatus.flags1 |= BS_FLAGS1_80;
+    } else {
+        gBattleStatus.flags1 &= ~BS_FLAGS1_80;
+    }
+    if (flags & BS_FLAGS1_800) {
+        gBattleStatus.flags1 |= BS_FLAGS1_800;
+    } else {
+        gBattleStatus.flags1 &= ~BS_FLAGS1_800;
+    }
```

Comment:
This is a great case for us allowing brace-less ifs I think; this form appears a lot in other places too. or we could macro it like `SET_FLAG(flags & BS_FLAGS_800, gBattleStatus.flags1, BS_FLAGS1_800);`